File: xis-server/src/main/java/one/xis/server/Request.java
package one.xis.server;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import lombok.Data;

import java.util.Map;

@Data
public class Request {

    @JsonDeserialize(using = DataDeserializer.class)
    private Map<String, String> data;
    private String clientId;
    private String userId;
    private String action;
    private String pageId;
    private String widgetId;
    private String nextPageId;
    private String nextWidgetId;
    private Map<String, String> parameters;
    private Map<String, String> pathVariables;
}

File: xis-javascript/src/main/js/classes/connect/Request.js
/**
 * @property {any} data
 * @property {string} clientId
 * @property {string} userId
 * @property {string} pageId
 * @property {string} widgetId 
 * @property {string} action
 * @property {string} nextPageId
 * @property {string} nextWidgetId 
*/
class Request {

    constructor() {
        this.data = {};
        this.clientId = '';
        this.userId = '';
        this.pageId = '';
        this.widgetId = '';
        this.action = '';
        this.nextPageId = '';
        this.nextWidgetId = '';
    }
}